cranelift/reader/src/isaspec.rs

```diff
@@ -92,7 +107,11 @@ where
             TestOption::Value(name, value) => match config.set(name, value) {
                 Ok(_) => {}
                 Err(SetError::BadName(name)) => {
-                    return Err(ParseOptionError::UnknownFlag { loc, name })
+                    return Err(ParseOptionError::UnknownValue {
+                        loc,
+                        name,
+                        value: value.to_string(),
+                    })
```

Comment:
I guess I don't see how changing the error type returned fixes the issue you mention in the PR description. Shouldn't `BadName` still map to `UnknownFlag` but the `UnknownValue` be used for some new case where the value isn't understood?

Reply:
Ah, ok, I think I see now; thanks for the explanation.